accepted/00000-reboot-required-flag.md

```diff
@@ -0,0 +1,59 @@
+## Feature Name: Reboot Required Flag for non-SUSE Distros
+**Start Date:** 23/11/2023
+
+### Summary
+Expand reboot required indication in Uyuni’s WebUI to systems running non-SUSE operating systems.
+
+### Motivation
+The feature of showing in the Web UI when a system managed by Uyuni requires a reboot is currently exclusive to SUSE products, it is not supported for systems running non-SUSE operating systems. The feature proposed here will bring a more comprehensive reboot notification capacity encompassing Ubuntu, RHEL, Rocky, Alma, Oracle Linux, and other non-SUSE distributions, enhancing the support for these distros within Uyuni.
+
+### Detailed Design
+
+#### Current Approach Used for SUSE Distros
+Currently, except for transactional systems (i.e. SLE Micro), Uyuni lacks a system attribute to indicate directly whether a given system requires a reboot at a given moment, the information is derived from other attributes: the time of the last boot and the packages installed in the system. If a package has the “reboot suggested/needed/required” characteristic, and it is installed in a system after its last boot, Uyuni then shows in the WebUI that a reboot is required for this system. It is interesting to note that Uyuni is sensitive only to package installations. If a package removal or any other change causes a reboot to be required, this will not be detected by Uyuni. The “reboot suggested/needed/required” characteristic of a package is also derived from other data that is only available for SUSE Distros.
+
+  
+#### Proposed Solution
+The proposed solution is to introduce an execution module that reads directly from the system whether a reboot is required, following specific approaches according to the OS family and version. This execution module will run together with package profile update and the result will be persisted in the database using the `reboot_needed` column at `suseminioninfo` table. During a minion startup event the flag will be set to `false` assuming that the reboot should not be necessary anymore as the system is starting.
+
+Although there are other possible reasons why a reboot may be necessary (a kernel parameter change, for instance), our current approach for SUSE distros is based only on packages/patch installation and seems to be enough for covering the most relevant use cases. Considering that a package profile update already runs after any package/patch change - even if it runs directly on the system - running the module together with package profile update should be enough for capturing when the reboot is required.
```

Comment:
> Considering that a package profile update already runs after any package/patch change - even if it runs directly on the system


This assertion is inaccurate. We do not execute a package profile update after each package or patch installation. It is only triggered in the event of substantial changes. Typically, we receive and apply the delta, updating the database directly.